api/src/main/java/org/openmrs/Encounter.java

```diff
@@ -492,16 +493,31 @@ public void setDiagnoses(Set<Diagnosis> diagnoses) {
 	/**
 	 * Basic property getter for conditions
 	 * 
-	 * @return conditions - set of conditions
+	 * @return all non-voided conditions
+	 * <strong>Should</strong> not get voided conditions
 	 * @since 2.4.0, 2.3.1
 	 */
 	public Set<Condition> getConditions() {
-		if (conditions == null) {
-			conditions = new LinkedHashSet<>();
-		}
```

Comment:
While the implementation here is simpler, it does potentially break existing code which relies on being able to do, say, `getConditions().add(newCondition)`.

Reply:
@ibacher you're right, but given that there is a `addCondition(Condition)` method, I guess we should assume that this is just bad practise no?

Reply:
@mks-d I suppose that's fair. And realistically, I don't expect there to be much actual code out there that uses this.